// openclaw/dist/commands-compact.runtime-Dop7frip.js
import { a as resolveSessionFilePath, o as resolveSessionFilePathOptions } from "./paths-CHZBIGhF.js";
import { s as resolveFreshSessionTotalTokens } from "./types-CoDcFuoc.js";
import { a as enqueueSystemEvent } from "./system-events-DNR0hK3x.js";
import { S as waitForEmbeddedAgentRunEnd, n as abortEmbeddedAgentRun, o as isEmbeddedAgentRunAbortableForCompaction } from "./runs-axTpTpGH.js";
import "./sessions-nxc5fSfR.js";
import { t as formatTokenCount } from "./token-format-D942KbWN.js";
import { n as compactEmbeddedAgentSession } from "./embedded-agent-Cv8lGIPa.js";
import { n as incrementCompactionCount } from "./session-updates-Cxc8v37L.js";
import { n as formatContextUsageShort } from "./status-message-C3AgWMec.js";
import "./status-BmbGiMcl.js";
export { abortEmbeddedAgentRun, compactEmbeddedAgentSession, enqueueSystemEvent, formatContextUsageShort, formatTokenCount, incrementCompactionCount, isEmbeddedAgentRunAbortableForCompaction, resolveFreshSessionTotalTokens, resolveSessionFilePath, resolveSessionFilePathOptions, waitForEmbeddedAgentRunEnd };
