// openclaw/dist/dispatch-acp-transcript.runtime--11uaO0E.js
import { v as resolveSessionAgentId } from "./agent-scope-CDZXADgT.js";
import { b as loadSessionStore, et as resolveSessionStoreEntry } from "./store-jR_nVe0n.js";
import { d as resolveStorePath } from "./paths-CHZBIGhF.js";
import { n as resolveAcpSessionCwd } from "./session-identifiers-C00NvSwu.js";
import "./sessions-nxc5fSfR.js";
import { c as persistAcpTurnTranscript } from "./attempt-execution-BhNnGXza.js";
//#region src/auto-reply/reply/dispatch-acp-transcript.runtime.ts
async function persistAcpDispatchTranscript(params) {
	const promptText = params.promptText.trim();
	const finalText = params.finalText.trim();
	if (!promptText && !finalText) return;
	const sessionAgentId = resolveSessionAgentId({
		sessionKey: params.sessionKey,
		config: params.cfg
	});
	const storePath = resolveStorePath(params.cfg.session?.store, { agentId: sessionAgentId });
	const sessionStore = loadSessionStore(storePath, { skipCache: true });
	const sessionEntry = resolveSessionStoreEntry({
		store: sessionStore,
		sessionKey: params.sessionKey
	}).existing;
	const sessionId = sessionEntry?.sessionId;
	if (!sessionId) throw new Error(`unknown ACP session key: ${params.sessionKey}`);
	await persistAcpTurnTranscript({
		body: promptText,
		transcriptBody: promptText,
		finalText,
		sessionId,
		sessionKey: params.sessionKey,
		sessionEntry,
		sessionStore,
		storePath,
		sessionAgentId,
		threadId: params.threadId,
		sessionCwd: resolveAcpSessionCwd(params.meta) ?? process.cwd(),
		config: params.cfg
	});
}
//#endregion
export { persistAcpDispatchTranscript };
